diff --git a/node_modules/patch-package/dist/getPackageResolution.js b/node_modules/patch-package/dist/getPackageResolution.js
index a4fa17f..7d82471 100644
--- a/node_modules/patch-package/dist/getPackageResolution.js
+++ b/node_modules/patch-package/dist/getPackageResolution.js
@@ -31,7 +31,7 @@ function getPackageResolution(_a) {
         var entries = Object.entries(appLockFile.object).filter(function (_a) {
             var k = _a[0], v = _a[1];
             return k.startsWith(packageDetails.name + "@") &&
-                v.version === installedVersion_1;
+                installedVersion_1.startsWith(v.version);
         });
         var resolutions = entries.map(function (_a) {
             var _ = _a[0], v = _a[1];
