diff --git a/node_modules/react-server-dom-webpack/cjs/react-server-dom-webpack.development.js b/node_modules/react-server-dom-webpack/cjs/react-server-dom-webpack.development.js
index eaf09fd..46f4986 100644
--- a/node_modules/react-server-dom-webpack/cjs/react-server-dom-webpack.development.js
+++ b/node_modules/react-server-dom-webpack/cjs/react-server-dom-webpack.development.js
@@ -49,7 +49,8 @@ function preloadModule(moduleData) {
     var entry = chunkCache.get(chunkId);
 
     if (entry === undefined) {
-      var thenable = __webpack_chunk_load__(chunkId);
+      // TODO: only preload here don't execute. Need's parcel support.
+      var thenable = module.bundle.root(chunkId);
 
       var resolve = chunkCache.set.bind(chunkCache, chunkId, null);
       var reject = chunkCache.set.bind(chunkCache, chunkId);
@@ -75,7 +76,7 @@ function requireModule(moduleData) {
     }
   }
 
-  var moduleExports = __webpack_require__(moduleData.id);
+  var moduleExports = module.bundle.root(moduleData.id);
 
   if (moduleData.name === '*') {
     // This is a placeholder value that represents that the caller imported this
